charts/datadog/templates/cluster-agent-confd-configmap.yaml

```diff
@@ -14,6 +14,9 @@ data:
 {{- end }}
 {{- if .Values.datadog.kubeStateMetricsCore.enabled }}
   kubernetes_state_core.yaml.default: |-
+{{- if .Values.datadog.kubeStateMetricsCore.useClusterCheckRunners }}
+    cluster_checks: true
```

Comment:
we also need to configure `skip_leader_election: true` in the instance field below when it's a cluster check. (see internal config for ref)